sandworm/lib/metadata.ml

```diff
@@ -0,0 +1,66 @@
+module Target = struct
+  type t =
+    | Aarch64_apple_darwin
+    | X86_64_apple_darwin
+    | X86_64_unknown_linux_musl
+  [@@deriving yojson]
+
+  let to_string = function
+    | Aarch64_apple_darwin -> "aarch64-apple-darwin"
+    | X86_64_apple_darwin -> "x86_64-apple-darwin"
+    | X86_64_unknown_linux_musl -> "x86_64-unknown-linux-musl"
+  ;;
+
+  let defaults = [ Aarch64_apple_darwin; X86_64_apple_darwin; X86_64_unknown_linux_musl ]
+end
+
+module Bundle = struct
+  type t =
+    { date : Ptime.date
+    ; targets : Target.t list
+    }
+
+  let create ~date targets = { date; targets }
+
+  let create_daily targets =
+    let date = Unix.time () |> Ptime.of_float_s |> Option.get |> Ptime.to_date in
+    create ~date targets
+  ;;
+
+  let get_data_string_from t =
+    let y, m, d = t.date in
+    Format.sprintf "%d-%02d-%02d" y m d
+  ;;
+end
+
+module Json = struct
+  type obj =
+    { date : float
```

Comment:
I'd avoid storing a `date` as float, if you depend on `Ptime` anyway.

Reply:
This issue is that if I have a `Ptime.t`, I don't know how to store it in JSON format... Do you have any idea how I could do it? Which format do you think would be better?

Reply:
Solved in a private discussion but the solution was to use custom Yojson de- and encoders that are then called by ppx_deriving_yojson.